hw/ip/pinmux/data/pinmux.hjson.tpl

```diff
@@ -294,41 +303,214 @@
+                }
+    },
+
+//////////////////////////
+// MIO PAD sleep mode   //
+//////////////////////////
+    { multireg: { name:     "MIO_PAD_SLEEP_STATUS",
+                  desc:     "Register indicating whether the corresponding pad is in sleep mode.",
+                  count:    "NMioPads",
+                  swaccess: "rw0c",
+                  hwaccess: "hrw",
+                  cname:    "MIO_PAD",
+                  fields: [
+                    { bits:   "0",
+                      name:   "EN",
+                      desc:   '''
+                              This register is set to 1 if the deep sleep mode of the corresponding
+                              pad has been enabled (!!MIO_OUT_SLEEP_MODE) upon deep sleep entry.
+                              The sleep mode of the corresponding pad will remain active until SW
+                              clears this bit.
+                              ''',
+                      resval: "1",
+                    }
+                  ]
+                }
+    },
+    { multireg: { name:     "MIO_PAD_SLEEP_REGWEN",
+                  desc:     "Register write enable for MIO sleep value configuration.",
+                  count:    "NMioPads",
+                  swaccess: "rw0c",
+                  hwaccess: "none",
+                  cname:    "MIO_PAD",
+                  fields: [
+                    { bits:   "0",
+                      name:   "EN",
+                      desc:   '''
+                              Register write enable bit.
+                              If this is cleared to 0, the corresponding !!MIO_OUT_SLEEP_MODE
+                              is not writable anymore.
+                              ''',
+                      resval: "1",
+                    }
+                  ]
+                }
+    },
+    { multireg: { name:         "MIO_PAD_SLEEP_MODE",
+                  desc:         '''Defines sleep behavior of the corresponding muxed pad.
+                                '''
+                  count:        "NMioPads",
+                  compact:      "false",
+                  swaccess:     "rw",
+                  hwaccess:     "hro",
+                  regwen:       "MIO_PAD_SLEEP_REGWEN",
+                  regwen_multi: "true",
+                  cname:        "OUT",
+                  fields: [
+                    { bits: "0",
+                      name: "EN",
+                      resval: 0,
+                      desc: '''
+                            Deep sleep mode enable.
+                            If this bit is set to 1 the corresponding pad will enable the sleep behavior
+                            specified in this register upon deep sleep entry, and the corresponding bit
+                            in !!MIO_PAD_SLEEP_STATUS will be set to 1.
+                            The pad remains in deep sleep mode until the corresponding bit in
+                            !!MIO_PAD_SLEEP_STATUS is cleared by SW.
```

Comment:
do you want to add a note either here or in the spec that if an always on peripherals is connected, it should just `EN` to 0? is that the idea?

Reply:
yeah that's the idea. let me add that to the description.